exporters/otlp/internal/transform/metric.go

```diff
@@ -234,24 +248,44 @@ func sink(ctx context.Context, in <-chan result) ([]*metricpb.ResourceMetrics, e
 // Record transforms a Record into an OTLP Metric. An ErrUnimplementedAgg
 // error is returned if the Record Aggregator is not supported.
 func Record(r export.Record) (*metricpb.Metric, error) {
-	switch a := r.Aggregation().(type) {
-	case aggregation.MinMaxSumCount:
-		return minMaxSumCount(r, a)
-	case aggregation.Sum:
-		return sum(r, a)
+	agg := r.Aggregation()
+	switch agg.Kind() {
+	case aggregation.MinMaxSumCountKind:
+		return minMaxSumCount(r, agg.(aggregation.MinMaxSumCount))
+
+	case aggregation.SumKind:
+		s, ok := agg.(aggregation.Sum)
+
+		if !ok {
```

Comment:
Is this ever a possibility, and if so can we add a unit test for it?

Reply:
Yes, I too wondered what the example would be where the record's aggregation kind was `aggregation.SumKind`, but the aggregator being used was not an `aggregation.Sum`?

Reply:
This shouldn't happen.

The reason this condition arises is to allow for a new kind of Aggregator e.g. that implements Sum but is not a known type (i.e., not `*sum.Aggregator`). If the Aggregator lies and then does not convert to the implied kind, this error will happen.

I've added tests for all of these error cases.